### PR TITLE
feat: support service monitor

### PR DESCRIPTION
**Pull request by howieyuen**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
refactor prometheus pod mixin to support both service monitor and pod minitor
```

### CODE REVIEW

**Review by Peefy — APPROVED** [review]
LGTM

**Review by SparkYuan — DISMISSED** [review]
lgtm

File: base/pkg/kusion_prometheus/mixins/monitor_mixin.k [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import base.pkg.kusion_models.kube.metadata
+import base.pkg.kusion_models.kube.protocol
+import base.pkg.kusion_models.kube.resource
+import base.pkg.kusion_models.kube.utils
+import base.pkg.kusion_prometheus.monitoring.v1 as monitoringv1
+
+mixin MonitorMixin for protocol.ServerProtocol:
```
Comment (amyXia1994) [inline base/pkg/kusion_prometheus/mixins/monitor_mixin.k:7]: please provide some schema documents for MonitoMixin and more line comments on the code.

Comment (howieyuen) [reply base/pkg/kusion_prometheus/mixins/monitor_mixin.k:7]: ok

**Review by amyXia1994 — APPROVED** [review]
lgtm

**Pull request merged**